client/src/pages/Register/PaymentStep.tsx

```diff
@@ -69,6 +94,44 @@ function PaymentStep(props: RegisterStepProps) {
     return response;
   };
 
+  const processResult = async (paymentType: PaymentType, payPalResponse?: OrderResponseBody) => {
+    let depositType: DepositType = 'none';
+
+    try {
+      const parsedDepositChoice = JSON.parse(depositChoice?.deposit);
+
+      depositType = parsedDepositChoice.name;
+    } catch (e) {
+      // nothing to do here
+    }
+
+    const initialPayment = {
+      registrationUUID: paymentStep.registrationUUID,
+      paymentType,
+      paymentData: {
+        type: depositType,
+        total,
+      },
+      payPalResponse,
+    };
+
+    dispatch(setPaymentInfo(initialPayment));
+    const result = await createInitialPayment(initialPayment);
+
+    if ('error' in result) {
+      throw new Error(`error creating payment: ${JSON.stringify(result)}`);
+    }
+
+    const { data } = result;
+
+    dispatch(setFinalStep(data));
+
+    history.push({
+      ...history.location,
+      pathname: history.location.pathname.replace('/payment', '/finished'),
```

Comment:
chance this to regex, or maybe create a special function that finds the correct url with a bit more care